libraries/c_sdk/standard/mqtt/src/iot_mqtt_subscription_container.c

```diff
@@ -0,0 +1,358 @@
+/*
+ * FreeRTOS MQTT SHIM V1.0.0
+ * Copyright(C) 2020 Amazon.com, Inc. or its affiliates.All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this softwareand associated documentation files(the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and /or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions :
+ *
+ * The above copyright noticeand this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://aws.amazon.com/freertos
+ * http ://www.FreeRTOS.org
+ */
+
+/**
+ * @file iot_mqtt_array_container.c
+ * @brief Implements array data structure functions for holding subscriptions and opeartions in the shim.
+ */
+/* The config header is always included first. */
+#include "iot_config.h"
+
+/* Standard includes. */
+#include <string.h>
+
+/* Error handling include. */
+#include "private/iot_error.h"
+
+/* MQTT internal includes. */
+#include "private/iot_mqtt_internal.h"
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Matches a packet identifier and order.
+ *
+ * @param[in] pSubscription Pointer to #_mqttSubscription_t.
+ * @param[in] pMatch Pointer to a #_packetMatchParams_t.
+ *
+ * @return `true` if the arguments match the subscription's packet info; `false`
+ * otherwise.
+ */
+static bool _packetMatch( _mqttSubscription_t * pSubscription,
+                          void * pMatch );
+
+/**
+ * @brief Matches a topic name (from a publish) with a topic filter (from a
+ * subscription).
+ *
+ * @param[in] pSubscription Pointer to #_mqttSubscription_t.
+ * @param[in] pMatch Pointer to a #_topicMatchParams_t.
+ *
+ * @return `true` if the arguments match the subscription topic filter; `false`
+ * otherwise.
+ */
+static bool _topicMatch( _mqttSubscription_t pSubscription,
+                         void * pMatch );
+
+/*-----------------------------------------------------------*/
+
+static bool _packetMatch( _mqttSubscription_t * pSubscription,
+                          _packetMatchParams_t * pMatch )
```

Comment:
The type seems to be different from where it is declared.

Reply:
Addressed

Reply:
Is still see a different type in line 53

Reply:
I still see the issue. Please fix